routers/api/v1/repo/search.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package repo
+
+import (
+	"net/http"
+
+	"code.gitea.io/gitea/modules/context"
+	code_indexer "code.gitea.io/gitea/modules/indexer/code"
+	"code.gitea.io/gitea/modules/setting"
+	api "code.gitea.io/gitea/modules/structs"
+	"code.gitea.io/gitea/routers/api/v1/utils"
+	"code.gitea.io/gitea/services/convert"
+)
+
+// CodeSearch Performs a code search on a Repo
+func CodeSearch(ctx *context.APIContext) {
+	// swagger:operation GET /repos/{owner}/{repo}/code_search repository repoCodeSearch
+	// ---
+	// summary: Performs a code search on a Repo
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo
+	//   type: string
+	//   required: true
+	// - name: keyword
+	//   in: query
+	//   description: the keyword the search for
+	//   type: string
+	// - name: language
+	//   in: query
+	//   description: filter results by language
+	//   type: string
+	// - name: match
+	//   in: query
+	//   description: only excact match
+	//   type: boolean
+	// - name: page
+	//   in: query
+	//   description: page number of results to return (1-based)
+	//   type: integer
+	// - name: keyword
+	//   in: limit
+	//   description: page size of results
+	//   type: string
+	// responses:
+	//   "200":
```

Comment:
There should be a `422` response because `http.StatusUnprocessableEntity` might be returned

Reply:
Still pending?

Reply:
Missed it. It is now added!